### PR TITLE
start getting modules to use shared import error code

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
Start to get modules to use the shared `missing_required_lib` error message. This error message builds a common error message in the case of an error when importing a required module. It includes information such as;

* The module name
* The host it is required on
* An optional reason if the module wishes to add this
* The Python executable that ran the module

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lots

### CODE REVIEW

File: lib/ansible/module_utils/basic.py [hunk 1]
```diff
@@ -727,10 +727,13 @@ def jsonify(data, **kwargs):
...
 
 
-def missing_required_lib(library):
+def missing_required_lib(library, reason=None):
     hostname = platform.node()
-    return "Failed to import the required Python library (%s) on %s's Python %s. Please read module documentation " \
-           "and install in the appropriate location." % (library, hostname, sys.executable)
+    msg = "Failed to import the required Python library (%s) on %s's Python %s." % (library, hostname, sys.executable)
+    if reason:
+        msg += "This is required %s." % reason
```
Comment (nitzmahone) [inline lib/ansible/module_utils/basic.py]: needs a space before the quote here too

**Review by nitzmahone — DISMISSED** [review]
one minor formatting thing, otherwise LGTM

File: lib/ansible/module_utils/basic.py (hunk 1 above)
Comment (jborean93) [reply lib/ansible/module_utils/basic.py]: Thanks for the pickup, will wait to see if the tests pass before making the chagnge

**Comment by maxamillion** [comment]
big fan 👍

**Review by maxamillion — APPROVED** [review]

File: lib/ansible/module_utils/basic.py (hunk 1 above)
Comment (jborean93) [reply lib/ansible/module_utils/basic.py]: Should be fixed now

File: lib/ansible/modules/crypto/acme/acme_challenge_cert_helper.py [hunk 2]
```diff
@@ -179,7 +182,7 @@ def main():
         ),
     )
     if not HAS_CRYPTOGRAPHY:
-        module.fail(msg='cryptography >= 1.3 is required for this module.')
+        module.fail_json(msg=missing_required_lib('cryptography >= 1.3'), exception=CRYPTOGRAPHY_IMP_ERR)
```
Comment (felixfontein) [inline lib/ansible/modules/crypto/acme/acme_challenge_cert_helper.py:185]: Thanks for spotting this! Will this PR be backported?

**Review by felixfontein — APPROVED** [review]
I've checked the crypto/acme modules, they look fine.

File: lib/ansible/modules/crypto/acme/acme_challenge_cert_helper.py (hunk 2 above)
Comment (jborean93) [reply lib/ansible/modules/crypto/acme/acme_challenge_cert_helper.py:185]: Unfortunately I won't be able to backport this PR as `missing_required_lib` was only added in the 2.8 cycle. Feel free to open another PR that fixes this and backport it. I can rebase this once that is done or just create a manual backport after this is merged.

Comment (jborean93) [reply lib/ansible/modules/crypto/acme/acme_challenge_cert_helper.py:185]: Commited #51795 which fixed this separate and have rebased this PR.

**Review by samdoran — APPROVED** [review]

**Comment by drzraf** [comment]
`$ python3 test/sanity/validate-modules/validate-modules -w --format json lib/ansible/modules/<foo>`
(eg: with `lib/ansible/modules/source_control/gitlab_project.py`)

Results:
`Exception attempting to import module for argument_spec introspection, 'cannot import name 'missing_required_lib''`

```

Traceback (most recent call last):
  File \"/ansible/test/sanity/validate-modules/module_args.py\", line 84, in get_argument_spec
    mod = imp.load_source('module', filename)
  File \"/usr/lib/python3.6/imp.py\", line 172, in load_source
    module = _load(spec)
  File \"<frozen importlib._bootstrap>\", line 684, in _load
  File \"<frozen importlib._bootstrap>\", line 665, in _load_unlocked
  File \"<frozen importlib._bootstrap_external>\", line 678, in exec_module
  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed
  File \"lib/ansible/modules/<foo>\", line 172, in <module>
    from ansible.module_utils.basic import missing_required_lib
ImportError: cannot import name 'missing_required_lib'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File \"test/sanity/validate-modules/validate-modules\", line 1125, in _validate_ansible_module_call
    spec, args, kwargs = get_argument_spec(self.path)
  File \"ansible/test/sanity/validate-modules/module_args.py\", line 90, in get_argument_spec
    reraise(AnsibleModuleImportError, AnsibleModuleImportError('%s' % e), sys.exc_info()[2])
  File \".local/lib/python3.6/site-packages/ansible/module_utils/six/__init__.py\", line 699, in reraise
    raise value.with_traceback(tb)
  File \"ansible/test/sanity/validate-modules/module_args.py\", line 84, in get_argument_spec
    mod = imp.load_source('module', filename)
  File \"/usr/lib/python3.6/imp.py\", line 172, in load_source
    module = _load(spec)
  File \"<frozen importlib._bootstrap>\", line 684, in _load
  File \"<frozen importlib._bootstrap>\", line 665, in _load_unlocked
  File \"<frozen importlib._bootstrap_external>\", line 678, in exec_module
  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed
  File \"lib/ansible/modules/<foo>\", line 172, in <module>
    from ansible.module_utils.basic import missing_required_lib
module_args.AnsibleModuleImportError: cannot import name 'missing_required_lib
```

**Comment by jborean93** [comment]
@drzraf `missing_required_lib` was only added in the devel branch and has not been included in a stable release just yet. Going by your traceback it seems like it's looking up an Ansible install in a Python's site-packages so not sure if this is based on devel or an older release.

**Pull request merged**